src/storage/exec/UpdateNode.h

```diff
@@ -364,12 +364,13 @@ class UpdateTagNode : public UpdateNode<VertexID> {
                         }
                         auto oi = indexKey(partId, vId, reader_, index);
                         if (!oi.empty()) {
-                            if (planContext_->env_->checkRebuilding(planContext_->spaceId_,
-                                                                    partId, indexId)) {
+                            auto iState = planContext_->env_->getIndexState(planContext_->spaceId_,
+                                                                            partId,
+                                                                            indexId);
+                            if (planContext_->env_->checkRebuilding(iState)) {
                                 auto deleteOpKey = OperationKeyUtils::deleteOperationKey(partId);
                                 batchHolder->put(std::move(deleteOpKey), std::move(oi));
-                            } else if (planContext_->env_->checkIndexLocked(planContext_->spaceId_,
-                                                                            partId, indexId)) {
+                            } else if (planContext_->env_->checkIndexLocked(iState)) {
```

Comment:
I have a question, Is there a gap here?
What if the state has changed during the source from line 372 to line 374?
for example : 
the iState == rebuilding when line:367
the statue changed from building to locked when line:374. but now,  the iStatue still rebuilding.

Reply:
in line 315 has handled.